cli/src/commands/utils.ts

```diff
@@ -23,33 +23,62 @@ export async function makeProviderFromNodeURLAndSecret(
     throw new Error('node_url is not defined. Please use --node_url to set one')
   }
 
-  var keyPair: KeyPair
-  if (
-    (globalOptions.wallet && !globalOptions.password) ||
-    (!globalOptions.wallet && globalOptions.password)
-  ) {
-    throw new Error('Both wallet and password must be provided together.')
+  let keyPair: KeyPair
+  let provider: Web3Provider
+
+  try {
+    keyPair = await KeyPair.fromEnv(KEY_ENV_NAME)
+    console.info('Initializing provider with SECRET_KEY from env\n')
+    provider = Web3Provider.fromRPCUrl(
+      globalOptions.node_url as string,
+      keyPair
+    )
+    return provider
+  } catch (error) {
+    console.warn(`Failed to initialize keyPair: ${error} \n`)
+  }
+
+  try {
+    if (!globalOptions?.secret_key) throw new Error()
+    keyPair = await KeyPair.fromPrivateKey(globalOptions.secret_key)
+    console.info('Initializing provider with SECRET_KEY from config file\n')
+    provider = Web3Provider.fromRPCUrl(
+      globalOptions.node_url as string,
+      keyPair
+    )
+    return provider
+  } catch (error) {
+    console.warn(`Failed to initialize keyPair: ${error} \n`)
   }
```

Comment:
I don't like this way of doing things
IMO, we should only try to load the keypair if we are in the right conditions such as:
```suggestion
  let keypair;

  const envSecretKey = process.env[KEY_ENV_NAME];

  if (envSecretKey) {
    keypair = await KeyPair.fromEnv(envSecretKey);
  } else if (config.secret_key) {
    keypair = await KeyPair.fromPrivateKey(config.secret_key);
  } else if (config.wallet_path && config.wallet_password) {
    keypair = await importFromYamlKeyStore(config.wallet_path, config.wallet_password);
  } else {
    throw new Error('No valid method to load keypair.');
  }
```

Also you are duplicating code for making a provider, this way should fix the issue as you will load the provider once you have a valid keypair. 
Note that it could be interesting to make a funciton specific to keypair loading to improve code readability

Reply:
My suggestion misses the try catch *inside the if statements* to provide a cleaner error message, but you have the idea !

Reply:
Hey thanks for your feedback, I'll look at implementing this!